Remove use of self instead of this

codeheroics opened the pull request:
Assigning `var self = this;` through the `Sticky.jsx` file served no purpose (and as `self` [is now used in browsers](https://twitter.com/rauschma/status/721992059863896065), it shadowed a global variable)

yahoocla commented:
Thank you for submitting this pull request, however I do not see a valid CLA on file for you. Before we can merge this request please visit https://yahoocla.herokuapp.com/ and agree to the terms. Thanks! :smile:

yahoocla commented:
CLA is valid!

redonkulus commented:
@codeheroics I agree with the `self` global issue but it does help with minification. We could change the name to something else, i.e. `var that = self` or something equivalent.

codeheroics commented:
While I do understand the rationale behind this, the use of `var that = this` bothers me, as it reduces code clarity - while reading it I thought the `var self = this` were used to preserve the `this` reference in some kind of closure, not for a module size gain of about 0.3/0.4ko before gzip.

src-code commented:
I tend to agree with @codeheroics - if minification were really a major concern, there's a lot of lower hanging fruit we should tackle (eg, overuse of `var`, or primitives like `true`/`false`, etc...)  Code clarity seems like a greater concern.

src-code commented:
@codeheroics There's been some updates to this project since your pull was opened, do you want to rebase and freshen it up (there may be some new uses of `self`), so we can get it merged?

codeheroics commented:
@src-code Sorry, a bit late in answering!

I just rebased my PR and fixed all conflicts / replaced all `self` occurrences.

However, I am no longer involved on the project I was working on which was using react-stickynode, which means I don't have a quick way of testing that I haven't broken anything with this rebase. Please do give the branch a quick check before merging it :)

src-code commented:
Thank you @codeheroics!
